[Pattern Draft] Circle Communities

michael-basil opened the pull request:
This pattern follows on from recent presentations with the ISC and many conversations.

This is still in draft while we allow space for final commentary before formally proposing it.

michael-basil committed:
feat: Add Circle Communities pattern

michael-basil committed:
feat: Add Jason Fisher acknowledgement

michael-basil committed:
fix: Typo

michael-basil committed:
fix: Typo

michael-basil committed:
feat: Add DevOps and DevSecOps callouts

billwestfall committed:
Update circle-communities.md

Move Bill W to author

michael-basil committed:
feat: Add Dojo Consortium

michael-basil committed:
feat: Update Thingy Farm link

michael-basil committed:
feat: Touch up

michael-basil committed:
feat: Touch up

michael-basil committed:
refactor: Review with Nick Basil

michael-basil committed:
fix: Cleanup

michael-basil committed:
refactor: Review with Nick Basil

michael-basil committed:
refactor: Acknowledgements

michael-basil committed:
feat: Add Alfred

jimmanzzullo committed:
Update circle-communities.md

michael-basil committed:
feat: Add Maturity Model relation

michael-basil committed:
feat: Add related pattern linkage

michael-basil committed:
feat: Add related pattern linkage

aphor committed:
characterize circle communities

michael-basil committed:
Merge pull request #2 from dojo-center/circle-communities_jeremy

characterize circle communities

aphor committed:
admit authorship

anuolu committed:
Update circle-communities.md

Edited the sentence:  Except for the duties of a facilitator, no distinction is enforced between participants. This invokes round-table equality which maximizes authenticity and diversity.

michael-basil committed:
Merge pull request #3 from dojo-center/anuolu-patch-1

Update circle-communities.md

michael-basil committed:
fix: Typo

michael-basil committed:
fix: Infuse Agile

michael-basil committed:
feat: Infuse Agile

michael-basil committed:
feat: Infuse Agile

michael-basil committed:
fix: Typo

michael-basil committed:
fix: Status start at Initial

michael-basil committed:
feat: Add Core Team pattern

tsadler1988 commented on patterns/2-structured/circle-communities.md:
````diff
@@ -0,0 +1,81 @@
+  * Engaged via [InnerSource Commons Foundation](https://innersourcecommons.org) in Slack at ```#dojo-center```
+  * Affiliated with the Mindset, Agile and InnerSource sub-communities within _InnerSource Dojo from SAP_
+* [InnerSource Dojo from SAP](https://www.youtube.com/watch?v=fXoVm5iTSCc)
+* _Unplugged InnerSource_ sessions within the _Reuse & InnerSource_ workgroup of [SAP Cross-Platform Architecture](https://community.sap.com/t5/technology-blogs-by-sap/cross-product-architecture-embracing-conway-s-law-for-better-software/ba-p/13648600)
````
```suggestion
* _Unplugged InnerSource_ sessions within the _Reuse & InnerSource_ workgroup of [SAP Cross-Platform Architecture](https://community.sap.com/t5/technology-blogs-by-sap/cross-product-architecture-embracing-conway-s-law-for-better-software/ba-p/13648600)
* [Guilds at the BBC](https://youtu.be/MyYgk-jdjSw?t=511)
```

We've done deeper work on communities - I'll see if I can find a better reference - but this might be an OK known instance anyway?

tsadler1988 commented on patterns/2-structured/circle-communities.md:
```diff
@@ -0,0 +1,81 @@
+* [Nick Basil](https://www.linkedin.com/in/nick-basil)
+* [Allison Kaufmann](https://www.linkedin.com/in/allisonkaufmann29)
+* [Dojo Consortium](https://dojoconsortium.org)
+* [Thingy Farm](https://thingyfarm.com/)
```
```suggestion
* [Thingy Farm](https://thingyfarm.com/)

# Alias

* [Community of Practice](https://en.wikipedia.org/wiki/Community_of_practice)
* [Guilds](https://engineering.atspotify.com/2014/03/spotify-engineering-culture-part-1/)
```

IMO this is very similar to Communities of Practice, and potentially Guilds too.

tsadler1988 reviewed (commented):
Big plus 1 for communities! A great collaboration tool in general, and specifically can help spread InnerSource practices in your organisation.

michael-basil committed:
feat: Add BBC instance

Co-authored-by: Tom Sadler <thomas.sadler@bbc.co.uk>

michael-basil committed:
feat: Add aliases

Co-authored-by: Tom Sadler <thomas.sadler@bbc.co.uk>

spier commented on patterns/2-structured/circle-communities.md:
```diff
@@ -0,0 +1,87 @@
+* [Jeremy McMillan](https://www.linkedin.com/in/jeremymcm)
+* [AnuOlu Balogun](https://www.linkedin.com/in/anuolu)
+
+## Acknowledgements
```
```suggestion
## Acknowledgments
```

We are using the US-English spelling here

spier commented on patterns/2-structured/circle-communities.md:
```diff
@@ -0,0 +1,87 @@
+* [Dojo Consortium](https://dojoconsortium.org)
+* [Thingy Farm](https://thingyfarm.com/)
+
+# Alias
```
```suggestion
## Alias
```

spier reviewed (commented):
Left some structural fixes first.

michael-basil committed:
fix: Typo

Co-authored-by: Sebastian Spier <github@spier.hu>

michael-basil committed:
fix: Typo

Co-authored-by: Sebastian Spier <github@spier.hu>

michael-basil committed:
fix: Relative links to patterns

michael-basil committed:
refactor: Patlet and Problem based on feedback

spier commented on patterns/1-initial/circle-communities.md:
```diff
@@ -0,0 +1,92 @@
+* [Nick Basil](https://www.linkedin.com/in/nick-basil)
+* [Allison Kaufmann](https://www.linkedin.com/in/allisonkaufmann29)
+* [Dojo Consortium](https://dojoconsortium.org)
+* [Thingy Farm](https://thingyfarm.com/)
```
In this section we typically acknowledge individuals for their indirect contributions to the pattern, not "organizations".

Not sure about Thingy Farm. Could not tell from the website what it is.

Also the Dojo Consortium does not fit out typical approach here.

michael-basil replied:
I'm going to remove these.  They are too much of a stretch for this it seems.  I also feel based on your comment that it would be more likely that Dojo Consortium didn't directly contribute as an organization and is more loosely collected set of DevOps / Agile learning enthusist niche professionals.  There may be interested parties to this pattern to support the introduction of InnerSource as a practice they introduce to organizations in time.  I'm happy to discuss this at some future point if someone approached me on it.

michael-basil replied:
I have removed these.

Please resolve if you agree this is done.

spier commented on patterns/1-initial/circle-communities.md:
```diff
@@ -0,0 +1,92 @@
+
+## Known Instances
+
+* Dojo Circle
```
Is the Dojo Circle an organization that applies this pattern to solve a problem that they faced while implementing InnerSource?

michael-basil replied:
I believe this has been removed from another commit.

If you agree please resolve.  

If not, please help me understand where this shows up in the refreshed version.

Trueblueaddie committed:
Update circle-communities.md

As discussed, context is the missing element not necessarily a structured approach to learning.

Trueblueaddie committed:
Update circle-communities.md

Trueblueaddie committed:
fix: typo

Trueblueaddie committed:
feat: updated framing

michael-basil committed:
Merge pull request #5 from mindset-dojo/Trueblueaddie-patch-1

feat: updated framing

michael-basil committed:
fix: trailing spaces

michael-basil committed:
fix: clarify related pattern linkage

michael-basil committed:
fix: clarify known instances

michael-basil committed:
fix: typos

spier commented on patterns/1-initial/circle-communities.md:
```diff
@@ -0,0 +1,92 @@
+
+Scenario 2: Your organization is decentralized and often stakeholders are unaware of other projects or programs. InnerSource has struggled to get engagement as the culture is highly dispersed and segmented across the organization.
+
+## Forces
```
The Forces section should describe what makes the problem difficult, and should not contain hints to the solution yet.

The 4 bullets from the Problem section ("Lack of Contextual Understanding" etc) could be used as Forces.

The bullets that are currently listed as forces ("Diversity of Participants" etc) could be moved to the Solution instead.

michael-basil replied:
I have taken another pass at the Problem, Forces and Solutions sections based on your feedback.

I also considered with fresh eyes some distinctions from other types of Communites of Practice.

Please advise @spier

spier replied:
These changes work for me. Nice job!
Let's leave these as they are.

spier commented on patterns/1-initial/circle-communities.md:
```diff
@@ -0,0 +1,92 @@
+* [Jeremy McMillan](https://www.linkedin.com/in/jeremymcm)
+* [AnuOlu Balogun](https://www.linkedin.com/in/anuolu)
+
+## Acknowledgments
```
These are a lot of acknowledgments :)
What is your goal with acknowledging all of them?
Did they all indirectly contribute to this pattern?

I am commenting on this as the combined list of authors/acknowledgements is easily the longest of all of our existing patterns. And while some (self-)promotion and name dropping is certainly fine, I would not want this to become too extensive.

michael-basil replied:
Each person in this list participated to help cultivate this pattern.   I would remind that Bill, AnuOlu, Jim and I presented on the core concept underlying this pattern to the InnerSource Commons in early-mid 2024.  Jeremy was instrumental just as much or more than other names on that list and arguably could be moved up a slot or two.

This is a highly socially collaborative pattern and the lists reflects that. 

I feel all of the author listings and acknowledgements are warranted in such a socially-engaged pattern.

There was no rule stated coming in about the length of the list and it appears this is just a divergence from the current expected norm.

If you feel strongly that you want to set limits here, I would state my objection as that seems to be against the spirit of spreading the patterns themselves and then ask for direct clarity on the limits and then I would make a modification to comply to move the pattern forward.

Please let me know on how to proceed and or if this comment can be resolved.

spier replied:
If you want to keep the whole list as is, that is fine with me as well.

Your comment above seems to refer mostly to the people in the Authors list. Does the same reasoning apply to the Acknowledgments list?

michael-basil replied:
I ensured that the authors were really involved in the creation of the language of the pattern.  This included conversations and they added their name by a commit in effect "signing" the pattern.   I woudn't sign someone else's name as an author.  The same was true for Addie recently.

Since I am leading the charge to carry this pattern through I did set the order of list based on how influential and supportive people were in this process.

For the acknowledgements list, I have had a conversation with each person though the level to which it would stick would vary.   

I believe the list is the way it should be when I review it now.  That said, if a desire and request was made to pare it down I could do it.  I prefer not to, but I feel far less strongly about it compared to the authors in comparison.

Please let me know how to proceed when you have a moment and or resolve if you feel the way this has been approached meets the spirit and or intention of the section.🙏

spier replied:
Let's leave all of them in then. :)

I guess I was also a bit confused because I had never heard about many of these people in the context of the ISC. Typically I have heard of people one way or the other, before they contribute to a pattern.

However I am just taking this as a way that the ISC is growing in reach, and that you found a way to bring some new people in. So that is amazing :)

Let's leave this as is ❤️

spier commented on patterns/1-initial/circle-communities.md:
```diff
@@ -0,0 +1,92 @@
+* **Continuous Engagement**: Sustained participation is crucial for the longevity and impact of the community.
+* **Knowledge Sharing**: Effective mechanisms for sharing knowledge and best practices are needed.
+
+## Solutions
```
Is it possible to be more specific about what a Circle Community around InnerSource would look like?

Currently the description is more a general "this is how to start a Circle Community/Guild/Community of Practice" and contains no specifics related to InnerSource.

michael-basil replied:
@spier - Very fair point ... I reflected on this and restated the key sections to clarify the distinction.

Please advise.

spier replied:
Yes, it is certainly better now. Let's leave this as is.

spier commented on patterns/1-initial/circle-communities.md:
```diff
@@ -0,0 +1,92 @@
+* [InnerSource Dojo from SAP](https://www.youtube.com/watch?v=fXoVm5iTSCc)
+* _Unplugged InnerSource_ sessions within the _Reuse & InnerSource_ workgroup of [SAP Cross-Platform Architecture](https://community.sap.com/t5/technology-blogs-by-sap/cross-product-architecture-embracing-conway-s-law-for-better-software/ba-p/13648600)
+* [Guilds at the BBC](https://youtu.be/MyYgk-jdjSw?t=511)
+* [Mindset Dojo Community](https://dojo.center)
```
```suggestion
```

Known Instances refer to organizations that have implemented this pattern.

This looks like it is a commercial offering from you where you want to teach people about this and other concepts? So in our definition this would not be a Known Instance and should be removed.

michael-basil replied:
Mindset Dojo Community is pivoting a bit.  Still in early infancy.   This is a bit of a more expansive conversation that I don't want to cloud this pattern itself.

Arguably the practitioners within Mindset Dojo may bring this pattern into organizations they work with and then those instances would have article / case study linkages directly.

In my new role starting on the 13th I am eager to consider bringing this pattern in when the timing and forces are in alignment.   

For example, based on what I already know there may be potential interest, but I don't know for certain.  

As such, I have removed as you requested.

If you agree this is now done, please resolve.

spier commented on patterns/1-initial/circle-communities.md:
```diff
@@ -0,0 +1,92 @@
+
+Implement a strategic approach to forming and nurturing Circle Communities that emphasizes clear objectives, inclusive participation and continuous improvement.  This can be achieved through the following parallel efforts:
+
+* **Define the Purpose**: Clearly articulate the purpose and goals of the Circle Community.  Ensure that these align with the broader mission of your organization or initiative.
```
Here I realize that I am not sure what this pattern is about, at its core.

Two theories:

Is this pattern about starting a **single** Circle Community around InnerSource?
And then that circle/guild will specifically try to increase InnerSource adoption in the org?

Or is the pattern about starting **multiple** Circle Communities (about various topics)?
And then practicing these circles/guilds will eventually lead to a certain mindset shift and change in collaboration approach, which is the basis for making InnerSource possible?

michael-basil replied:
I addressed this in another comment near the end of the file with some tangentially connected (in my mind) questions.

Please resolve or let me know if you have more questions (here or there no difference to me).

spier commented on patterns/1-initial/circle-communities.md:
```diff
@@ -0,0 +1,92 @@
+
+## Patlet
+
+InnerSource adoption in organizations is slow due to limited understanding and engagement. Circle Communities foster a culture of collaboration and continuous learning, which are essential to InnerSource. These communities provide a platform for synchronous conversations, where knowledgeable and confident members can engage in active listening and help build the necessary connections and understa ...
```
Can we try to get this down to two sentences?
1st sentence / the problem
2nd sentence / the solution and how it contributes to solving the problem

it is hard but it is also a good exercise as it forces us to answer the question specifically:
How does this solution help us to improve the given InnerSource challenge?

michael-basil replied:
I placed a fresh comment on the file with the refactored language.

Please advise (there ideally).

spier commented on patterns/1-initial/circle-communities.md:
```diff
@@ -0,0 +1,92 @@
+
+## Context
+
+Scenario 1: Few InnerSource champions exist within your heavily siloed organization. The practice has struggled to build sustainable engagement as the culture is not hospitable to open collaboration and innovation.
```
```suggestion
Scenario 1: Few people with extensive InnerSource knowledge exist within your heavily siloed organization. The practice has struggled to build sustainable engagement as the culture is not hospitable to open collaboration and innovation.
```

I would not use the word "champion" here, as we are trying to establish ambassador/champion as a pattern and term in its own right.

Therefore I would rather describe what is meant here. I tried but not sure if I got it right?

spier commented on patterns/1-initial/circle-communities.md:
```diff
@@ -0,0 +1,92 @@
+
+## Resulting Context
+
+Embracing this pattern helps InnerSource programs become vibrant, effective and sustainable. Participants will feel more connected and engaged, leading to increased knowledge sharing, innovation and collective growth.
```
I cannot clearly see the connection between the Solution and the Resulting Context.

Can we describe more specifically how the Solution is improving upon the listed Problem?

michael-basil replied:
I placed a fresh comment on the file with the refactored language.

Please advise (there ideally).

spier commented on patterns/1-initial/circle-communities.md:
```diff
@@ -0,0 +1,92 @@
+* [Dedicated Community Leader](../2-structured/dedicated-community-leader.md)
+* [Core Team](../2-structured/core-team.md)
+* [Maturity Model](../2-structured/maturity-model.md)
+
```
```suggestion
```

Moving this section further down.

spier commented on patterns/1-initial/circle-communities.md:
```diff
@@ -0,0 +1,92 @@
+* _Unplugged InnerSource_ sessions within the _Reuse & InnerSource_ workgroup of [SAP Cross-Platform Architecture](https://community.sap.com/t5/technology-blogs-by-sap/cross-product-architecture-embracing-conway-s-law-for-better-software/ba-p/13648600)
+* [Guilds at the BBC](https://youtu.be/MyYgk-jdjSw?t=511)
+* [Mindset Dojo Community](https://dojo.center)
+
```
```suggestion

## Related Patterns

* [Dedicated Community Leader](../2-structured/dedicated-community-leader.md)
* [Core Team](../2-structured/core-team.md)
* [Maturity Model](../2-structured/maturity-model.md)

```

spier commented on patterns/1-initial/circle-communities.md:
```diff
@@ -0,0 +1,92 @@
+* [Dojo Consortium](https://dojoconsortium.org)
+* [Thingy Farm](https://thingyfarm.com/)
+
+## Alias
```
Which term for this concept is most frequently used in the industry?

Just a single personal data point:
I have heard of guilds and community of practice frequently. I had never heard the term circle communities before.

spier replied:
Also would Dojo be yet another alias for this?
Or is a Dojo more focused on the collaborative practicing of things?

michael-basil replied:
1. "the industry" is pretty subjective ...

If we are talking broadly speaking then if I am taking "the industry" to include communication and change experts when I talk about a "Circle" community they get an idea of where this is going.

I would consider Circle Communities as a subset of Communities of Practice.  I have ammended several of the sections per other feedback which should suggest where the distinctions are to merit such a separate designation.

I also have conversations with my new employer about a Guild they have (for Agile) that lacks some of the form and flow of a Circle and they are excited to hear more.   I would say that a Circle Community itself is inherently aligned with InnerSource Principles wherase an Agile Circle for example inherets the principles of InnerSource into it's DNA on inception unknowingly.

Moreover, Circle Communties support InnerSource adoption from the bottom up and middle out aligned to top-down or central initiatives whenever they exist and those groups are supportive and willing to engage (always invited).  Support it in specific ways not common to Communities of Practice in general.

2. I would not call a Circle an alias for Dojo.  

The Dojo metaphor and model carries much different distinctions and social-learning system advanced support.  Not all Dojos have Circles, though all Circles could theoretically (if the group desired) spawn and connect to a Dojo learning approach.  I would say though that a Circle could spawn different models as well.  The SAP Dojo featured Circles whereas the Microsoft DevOps Dojo likely did not (though I cannot be sure as I never met anyone directly part of that unfortuanately, at least not yet).

I decided to close the PR we had for a Dojo pattern to allow the focus to center here on the far more important pattern and to consider revisiting if a Dojo pattern made sense, though there would need to be interest and appetite from the InnerSource Commons Community for this (AND IDEALLY participation from Microsoft DevOps Dojo key person(s)). 

The Circle pattern does have affinity and interest from at least one important member who is also listed as an author (you will know the name(s) when you encounter them).  Had that not existed I would have closed the pattern.

I would also add that this Circle Communities pattern is ideal for socially oriented DevOps groups and Agilists.   Additionally, I would like to have a conversation with Cloud Security Office Hours sometime this year after it is published.  Bill and I have presented there in the past and I'm sure he'd be interested and I would love it if someone or multiple people who are with ISC as formal members would participate - but that is not the key point here (just planting an invitational seed for later).  I mention this because I believe outreach into other communities is a conversation I have heard Russ talk about a few times.

3. As far as if this pattern is a single or multiple instance?  

Yes.  More like WHEN and DEPENDING ON what.

Consider it a pattern of flow versus control.   

I have seen it where Circles spawn short and medium lived other Circles or long lived Circle-style collaboration zones in different areas of a business unit after experiencing it.

I say that it starts and ends with one circle.  This is from one point of view though.  Just like a repository codebase there may be multiple "centrals" depending on branching, forking, etc.  Same concept here.

Apologies for the length of this I felt it may be of use to give an expansive answer.  I hope you'll forgive me if it is taxing!  

🙏🌿

spier replied:
There is a lot to unpack here :) I will leave this comment open for future reference.

spier commented on patterns/2-structured/dedicated-community-leader.md:
```diff
@@ -56,6 +56,11 @@ A community leader with the properties described above will lend a face and embo
+
+* [Circle Communities](https://patterns.innersourcecommons.org/p/circle-communities)
+* [Agile InnerSource Dojo](https://patterns.innersourcecommons.org/p/agile-innersource-dojo)
+
```
Let's handle the related patterns in a separate PR.

michael-basil replied:
Removed.

michael-basil replied:
If you agree this is done, please resolve.

spier commented on patterns/2-structured/maturity-model.md:
```diff
@@ -208,6 +208,12 @@ Prior to adopting InnerSource as a working model, teams are aware of the
+
+* [Dedicated Community Leader](https://patterns.innersourcecommons.org/p/dedicated-community-leader)
+* [Circle Communities](https://patterns.innersourcecommons.org/p/circle-communities)
+* [Agile InnerSource Dojo](https://patterns.innersourcecommons.org/p/agile-innersource-dojo)
```
Let's handle the related patterns in a separate PR.

michael-basil replied:
I removed the section.  Not sure if there is a better way to remove the file from the PR overall.

michael-basil replied:
If you agree this is done, please resolve.

spier replied:
The file has disappeared from the PR already, as it did not contain any changes any more. 👍

spier reviewed (commented):
@michael-basic I have left various comments inline. As you can tell I am still struggling a bit to understand the direct link between the described concept and InnerSource.

I hope you don't perceive my comments as overly critical. I am only trying to understand how this patterns will help people to improve their InnerSource practices. Therefore I am asking so many questions :)

michael-basil committed:
fix: Cleanup acknowledgements - remove organizations

michael-basil committed:
fix: Remove per request

michael-basil committed:
fix: Requested cleanup

Co-authored-by: Sebastian Spier <github@spier.hu>

michael-basil committed:
fix: To be handled in separate PR

michael-basil committed:
fix: To be handled in separate PR

michael-basil committed:
fix: Cleanup language to replace champion term

Co-authored-by: Sebastian Spier <github@spier.hu>

michael-basil committed:
fix: Cleanup requested

Co-authored-by: Sebastian Spier <github@spier.hu>

michael-basil committed:
fix: Clarify Patlet structure

michael-basil commented on patterns/1-initial/circle-communities.md:
```diff
@@ -0,0 +1,89 @@
+
+## Patlet
+
+InnerSource adoption is slow in organizations due to limited understanding, engagement, and contextual relevance. Circle Communities address this by fostering synchronous conversations that build connections, close knowledge gaps, and cultivate collaboration and continuous learning.
```
@spier - I refactored the Patlet section as requested.

Please advise.

michael-basil committed:
refactor: Restructure and augment primary sections

michael-basil committed:
refactor: Ground key sections into InnerSource specific aspects

michael-basil committed:
refactor: Connect Resulting Context to Solution

michael-basil commented on patterns/1-initial/circle-communities.md:
```diff
@@ -0,0 +1,109 @@
+* **Leverage Collaborative Tools**: Use tools like Slack or GitHub to facilitate communication, documentation, and privacy-preserving insights sharing.
+* **Foster Continuous Improvement**: Solicit regular feedback to refine the community's activities and structure.
+
+## Resulting Context
```
@spier - I refactored the Resulting Context section as requested.

Please advise.

michael-basil committed:
refactor: Resulting Context

michael-basil committed:
fix: Forces should not have solution item

michael-basil committed:
refactor: Make clear call-out for dedicated leadership

spier reviewed (approved):
Great job with the last round of changes.

Let's get this merged, and try to find organizations that have implemented this pattern. Hopefully those orgs can then help to harden this pattern even further, and possibly make the connection between Circle Communities and InnerSource even more clear.

Thanks again for your patience and rigor in working through this PR with me.
